Statistics -- update added

renukamanavalan opened the pull request:
<!--
     Please make sure you've read and understood our contributing guidelines:
     https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

     ** Make sure all your commits include a signature generated with `git commit -s` **

     If this is a bug fix, make sure your description includes "fixes #xxxx", or
     "closes #xxxx" or "resolves #xxxx"

     Please provide the following information:
-->

#### Why I did it

#### How I did it

#### How to verify it

#### Which release branch to backport (provide reason below if selected)

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

#### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

#### Link to config_db schema for YANG module changes
<!--
Provide a link to config_db schema for the table for which YANG model
is defined
Link should point to correct section on https://github.com/Azure/SONiC/wiki/Configuration.
-->

#### A picture of a cute animal (not mandatory but encouraged)

renukamanavalan committed:
Updated per new API in swsscommon

renukamanavalan committed:
compile fix

renukamanavalan committed:
Merge remote-tracking branch 'upstream/statistics' into statistics

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
compile fix

renukamanavalan committed:
connect to DB on demand

renukamanavalan committed:
syntax bug

renukamanavalan committed:
try

renukamanavalan committed:
minor

renukamanavalan committed:
added DB

renukamanavalan committed:
Adding DB

renukamanavalan committed:
include DB

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
fix

renukamanavalan committed:
minor

renukamanavalan committed:
telemetry does not initiate hearbeat events, as eventd does that

renukamanavalan committed:
skip heartbeat when q is non-empty

renukamanavalan committed:
block heartbeat

renukamanavalan committed:
compile

renukamanavalan committed:
compile

renukamanavalan committed:
drop SyncResponse

renukamanavalan committed:
minor

renukamanavalan committed:
debug

renukamanavalan committed:
minor

renukamanavalan committed:
debug

renukamanavalan committed:
debug

renukamanavalan committed:
debug

renukamanavalan committed:
debug

renukamanavalan committed:
debug

renukamanavalan committed:
debug

renukamanavalan committed:
dropped DEBUG code

Commit:
Fix pipeline.

ganglyu committed:
Merge pull request #8 from ganglyu/fix_pipeline

Fix pipeline.

renukamanavalan committed:
Pass heartbeat param to swsscommon lib

renukamanavalan committed:
syntax

renukamanavalan committed:
syntax

zbud-msft commented on gnmi_server/client_subscribe.go:
```diff
@@ -247,8 +247,9 @@ func (c *Client) recv(stream gnmipb.GNMI_SubscribeServer) {
-func (c *Client) send(stream gnmipb.GNMI_SubscribeServer) error {
+func (c *Client) send(stream gnmipb.GNMI_SubscribeServer, dc sdc.Client) error {
 	for {
+        var val *sdc.Value
```
Fix indentations as well as changing spaces to tabs

zbud-msft commented on gnmi_server/client_subscribe.go:
```diff
@@ -262,12 +263,14 @@ func (c *Client) send(stream gnmipb.GNMI_SubscribeServer) error {
 		var resp *gnmipb.SubscribeResponse
-		switch v := items[0].(type) {
+
+        switch v := items[0].(type) {
```
Indentation/tabs

zbud-msft commented on gnmi_server/client_subscribe.go:
```diff
@@ -262,12 +263,14 @@ func (c *Client) send(stream gnmipb.GNMI_SubscribeServer) error {
+        switch v := items[0].(type) {
 		case sdc.Value:
-			if resp, err = sdc.ValToResp(v); err != nil {
+            if resp, err = sdc.ValToResp(v); err != nil {
```
indentation/tabs

zbud-msft commented on gnmi_server/client_subscribe.go:
```diff
@@ -262,12 +263,14 @@ func (c *Client) send(stream gnmipb.GNMI_SubscribeServer) error {
 				c.errors++
 				return err
 			}
+            val = &v;
```
same

zbud-msft commented on sonic_data_client/events_client.go:
```diff
@@ -48,48 +87,192 @@ func NewEventClient(paths []*gnmipb.Path, prefix *gnmipb.Path, logLevel int) (Cl
+                cnt += 1
+            }
+        }
+        evtc.counters[LATENCY] = (uint64) (total/cnt/1000/1000)
```
Can we add check for cnt to not be 0, otherwise it'll be divide by 0 crash

renukamanavalan replied:
Very good catch! Thank you! Fixed.

zbud-msft commented on sonic_data_client/events_client.go:
```diff
@@ -48,48 +87,192 @@ func NewEventClient(paths []*gnmipb.Path, prefix *gnmipb.Path, logLevel int) (Cl
+        tmp_counters := make(map[string]uint64)
+
+        // compute latency
+        if evtc.last_latency_full {
```
When are you setting back to false?

zbud-msft replied:
Why are we checking twice, we call compute latency on 157?

renukamanavalan replied:
Set to False in NewEventClient.

Missed to replace the code with the function.

renukamanavalan committed:
review comments

Commit:
Add log for authentication and add unit test.

Signed-off-by: Gang Lv ganglv@microsoft.com

Commit:
Improve coding style.

ganglyu committed:
Add log for authentication and add unit test. #12

Signed-off-by: Gang Lv ganglv@microsoft.com

Why I did it
We need to enhance authentication and authorization for sonic-gnmi, and the first step is to add log.

How I did it
Add log for auth user and roles.

How to verify it
Run unit test.

renukamanavalan committed:
Merge pull request #2 from sonic-net/master

Merge master into stats branch
